src/System.Memory/src/System/SpanHelpers.cs

```diff
@@ -22,26 +22,14 @@ internal static partial class SpanHelpers
         /// </summary>
         public static unsafe void CopyTo<T>(ref T dst, int dstLength, ref T src, int srcLength)
         {
-            IntPtr srcMinusDst = Unsafe.ByteOffset<T>(ref dst, ref src);
-            bool srcGreaterThanDst = (sizeof(IntPtr) == sizeof(int)) ? srcMinusDst.ToInt32() >= 0 : srcMinusDst.ToInt64() >= 0;
-            IntPtr tailDiff;
+            const long srcStart = 0;
+            long srcEnd = srcStart + (long)Unsafe.ByteOffset(ref src, ref Unsafe.Add(ref src, srcLength));
 
-            if (srcGreaterThanDst)
-            {
-                // If the start of source is greater than the start of destination, then we need to calculate
-                // the different between the end of destination relative to the start of source.
-                tailDiff = Unsafe.ByteOffset<T>(ref Unsafe.Add<T>(ref dst, dstLength), ref src);
-            }
-            else
-            {
-                // If the start of source is less than the start of destination, then we need to calculate
-                // the different between the end of source relative to the start of destunation.
-                tailDiff = Unsafe.ByteOffset<T>(ref Unsafe.Add<T>(ref src, srcLength), ref dst);
-            }
+            long dstStart = (long)Unsafe.ByteOffset(ref src, ref dst);
```

Comment:
It would be nice to do the math using native integer arithmetic to make it perform well on 32-bit platforms. Either special case it for `sizeof(IntPtr) == sizeof(int))` or use a trick `byte*` instead of `long`, even though it is not really a pointer.